src/modules/attempts/attempts.service.ts

```diff
@@ -603,7 +623,54 @@ export class AttemptsService {
     }
   }
 
+  /**
+   * Calculates the final score for an attempt after manual review of subjective questions.
+   * 
+   * This is used after manual review when all subjective questions have been scored.
+   * 
+   * @param attemptId - The ID of the test attempt to calculate final score for
+   * @param authContext - Authentication context for tenant/organization filtering
+   * @returns Promise<number> - The final score as a percentage
+   */
   private async calculateFinalScore(attemptId: string, authContext: AuthContext): Promise<number> {
+    const { answers, totalMarks } = await this.getAttemptAnswersWithMarks(attemptId, authContext);
+    
+    let totalScore = 0;
+    for (const answer of answers) {
+      totalScore += answer.score || 0;
+    }
+
+    return totalMarks > 0 ? (totalScore / totalMarks) * 100 : 0;
+  }
+
+  /**
+   * Calculates the total possible marks for an attempt by summing up marks from all questions.
+   * 
+   * This function retrieves all questions in the attempt and sums their individual marks
+   * to determine the maximum possible score for the attempt.
+   * 
+   * @param attemptId - The ID of the test attempt to calculate total marks for
+   * @param authContext - Authentication context for tenant/organization filtering
+   * @returns Promise<number> - The total possible marks for the attempt
+   */
+  private async calculateTotalMarks(attemptId: string, authContext: AuthContext): Promise<number> {
+    const { totalMarks } = await this.getAttemptAnswersWithMarks(attemptId, authContext);
+    return totalMarks;
+  }
+
+  /**
+   * Retrieves all answers for an attempt along with the total possible marks.
+   * 
+   * This function:
+   * 1. Fetches all user answers for the specified attempt
+   * 2. For each answer, retrieves the associated question to get its marks
+   * 3. Calculates the total possible marks by summing all question marks
+   * 
+   * @param attemptId - The ID of the test attempt
+   * @param authContext - Authentication context for tenant/organization filtering
+   * @returns Promise<{answers: TestUserAnswer[], totalMarks: number}> - Object containing answers and total marks
+   */
+  private async getAttemptAnswersWithMarks(attemptId: string, authContext: AuthContext): Promise<{ answers: TestUserAnswer[], totalMarks: number }> {
```

Comment:
@vinayak-patil test has marks right? why we are getting questions in the attempts and calculate marks?